### PR TITLE
Pull RecordServiceAttachmentUploadTest out of RecordServiceUploadsUpdatesDownloadsTest

### PR DESCRIPTION
**Pull request by bitPogo**
## Description

This is the 3rd of 3 PRS, which breaks the RecordServiceUploadsUpdatesDownloadsTest down.
Currently RecordServiceUploadsUpdatesDownloadsTest is not in a proper state.
This is a successor of #66 and depends on #71.

## Motivation and Context

Currently RecordServiceUploadsUpdatesDownloadsTest is not in a proper state.

## How is it being implemented?

This breaks out the update part out of the RecordServiceUploadsUpdatesDownloadsTest.
It removes internal legacy methods, which are not longer need and removes the corresponding tests as well.

## How Has This Been Tested?

This is a rewrite of tests which check the upload attachment logic.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have updated the changelog accordingly.
- [ ] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

File: sdk-core/src/main/java/care/data4life/sdk/RecordService.kt [hunk 1]
```diff
@@ -124,11 +124,6 @@ class RecordService internal constructor(
             )
     )
 
-    @Deprecated("Deprecated with version v1.9.0 and will be removed in version v2.0.0")
```
Comment (wmontwe) [inline sdk-core/src/main/java/care/data4life/sdk/RecordService.kt:127]: 👍

File: sdk-core/src/main/java/care/data4life/sdk/RecordService.kt [hunk 2]
```diff
@@ -913,13 +908,9 @@ class RecordService internal constructor(
         val resource = record.resource
 
         if (!fhirAttachmentHelper.hasAttachment(resource)) return record
-        val attachments = fhirAttachmentHelper.getAttachment(resource) as List<Any?>?
+        val attachments = fhirAttachmentHelper.getAttachment(resource)
                 ?: return record
 
-        if (record.attachmentsKey == null) {
```
Comment (wmontwe) [inline sdk-core/src/main/java/care/data4life/sdk/RecordService.kt:919]: Why is this removed? Isn't there a case the ther is no attachment key yet?

Comment (bitPogo) [reply sdk-core/src/main/java/care/data4life/sdk/RecordService.kt:919]: It is actually not removed, it is just moved (as mentioned in the PR before) into `resolveAttachmentKey`, which is called by `uploadAttachmentsOnDemand `.

**Review by wmontwe — APPROVED** [review]
👍

File: sdk-core/src/main/java/care/data4life/sdk/RecordService.kt (hunk 2 above)
Comment (wmontwe) [reply sdk-core/src/main/java/care/data4life/sdk/RecordService.kt:919]: 👍
